spec/flipper/api/action_spec.rb

```diff
@@ -25,35 +25,72 @@ def delete
     end
   }
 
-  it "won't run method that isn't whitelisted" do
-    fake_request = Struct.new(:request_method, :env, :session).new("NOOOOPE", {}, {})
-    action = action_subclass.new(flipper, fake_request)
-    expect {
-      action.run
-    }.to raise_error(Flipper::Api::RequestMethodNotSupported)
-  end
+  describe 'https verbs' do
 
-  it "will run get" do
-    fake_request = Struct.new(:request_method, :env, :session).new("GET", {}, {})
-    action = action_subclass.new(flipper, fake_request)
-    expect(action.run).to eq([200, {}, "get"])
-  end
+    it "won't run method that isn't whitelisted" do
+      fake_request = Struct.new(:request_method, :env, :session).new("NOOOOPE", {}, {})
+      action = action_subclass.new(flipper, fake_request)
+      expect {
+        action.run
+      }.to raise_error(Flipper::Api::RequestMethodNotSupported)
+    end
 
-  it "will run post" do
-    fake_request = Struct.new(:request_method, :env, :session).new("POST", {}, {})
-    action = action_subclass.new(flipper, fake_request)
-    expect(action.run).to eq([200, {}, "post"])
-  end
+    it "will run get" do
+      fake_request = Struct.new(:request_method, :env, :session).new("GET", {}, {})
+      action = action_subclass.new(flipper, fake_request)
+      expect(action.run).to eq([200, {}, "get"])
+    end
 
-  it "will run put" do
-    fake_request = Struct.new(:request_method, :env, :session).new("PUT", {}, {})
-    action = action_subclass.new(flipper, fake_request)
-    expect(action.run).to eq([200, {}, "put"])
+    it "will run post" do
+      fake_request = Struct.new(:request_method, :env, :session).new("POST", {}, {})
+      action = action_subclass.new(flipper, fake_request)
+      expect(action.run).to eq([200, {}, "post"])
+    end
+
+    it "will run put" do
+      fake_request = Struct.new(:request_method, :env, :session).new("PUT", {}, {})
+      action = action_subclass.new(flipper, fake_request)
+      expect(action.run).to eq([200, {}, "put"])
+    end
+
+    it "will run delete" do
+      fake_request = Struct.new(:request_method, :env, :session).new("DELETE", {}, {})
+      action = action_subclass.new(flipper, fake_request)
+      expect(action.run).to eq([200, {}, "delete"])
+    end
   end
 
-  it "will run delete" do
-    fake_request = Struct.new(:request_method, :env, :session).new("DELETE", {}, {})
-    action = action_subclass.new(flipper, fake_request)
-    expect(action.run).to eq([200, {}, "delete"])
+  describe '#json_error_response' do
+    describe ":feature_not_found" do
+
+      it 'locates and serializes error correctly' do
+        action = action_subclass.new({}, {})
+        response = catch(:halt) do
+          action.json_error_response(:feature_not_found, 404)
+        end
+        expect(response).to eq([404, {"Content-Type"=>"application/json"}, ["{\"code\":1,\"message\":\"Feature not found.\",\"more_info\":\"\"}"]])
+      end
```

Comment:
wasn't quite sure if this is good way to test these, but I wanted to make sure we could test all the expected functionality of render_error_json, let me know if you'd rather test this another way